src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -219,7 +219,10 @@ else if (line[after] == ' ') {
      */
     private void processSingleGeneric(DetailAST ast, int[] line, int after) {
         final char charAfter = Character.toChars(line[after])[0];
-        if (isGenericBeforeMethod(ast) || isGenericBeforeCtor(ast)) {
+        if (isGenericBeforeMethod(ast)
+                // isGenericBeforeNew() needn't be checked as the ast which satisfies it,
+                // will also satisfy isGenericBeforeCtorInvocation().
```

Comment:
This is because `isGenericBeforeCtorInvocation()` encompasses the first-operand of the and-statement in `isGenericBeforeNew()` as the first-operand of its or-statement.
Inclusion of both will result in mutant.

Reply:
@sktpy , please remove this comment, I  do not understand it. isGenericBeforeNew is not existing.

Reply:
Done.
Previously `isGenericBeforeCtor()` (which is now called `isGenericBeforeNew()`) was checked here, but since it can now be covered with `isGenericBeforeCtorInvocation()`, had added that comment just in case that if logic gets changed in future in a way that `isGenericBeforeCtorInvocation()` wouldn't be able to satisfy ast which satisfies `isGenericBeforeNew()`